### PR TITLE
Winding order fix

### LINKED ISSUE #17683: Fix winding order in concentric circle mesh
## Reason
In order to reliably extrude a `ConcenticCircleMeshGenerator` mesh the element winding order needs to be fixed.

## Design
Painstakingly go through element creation and flip winding order where necessary. Also add a test AuxKernel to check the winding order.

## Impact
Improved mesh generation

### PR DESCRIPTION
**Pull request by dschwen**
Closes #17683

### CODE REVIEW

File: framework/src/meshgenerators/ConcentricCircleMeshGenerator.C [hunk 1]
```diff
@@ -383,6 +386,7 @@ ConcentricCircleMeshGenerator::generate()
 
   while (index < limit)
   {
+    // ?
```
Comment (GiudGiud) [inline framework/src/meshgenerators/ConcentricCircleMeshGenerator.C:389]: ?

File: framework/src/meshgenerators/ConcentricCircleMeshGenerator.C [hunk 2]
```diff
@@ -653,6 +659,7 @@ ConcentricCircleMeshGenerator::generate()
       {
         while (index < limit)
         {
+          // remove -> crash
```
Comment (GiudGiud) [inline framework/src/meshgenerators/ConcentricCircleMeshGenerator.C:662]: but why

File: test/tests/meshgenerators/concentric_circle_mesh_generator/concentric_circle_mesh2.i [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [Mesh]
-  [./ccmg]
```
Comment (GiudGiud) [inline test/tests/meshgenerators/concentric_circle_mesh_generator/concentric_circle_mesh2.i:2]: +1

File: framework/src/meshgenerators/ConcentricCircleMeshGenerator.C (hunk 2 above)
Comment (dschwen) [reply framework/src/meshgenerators/ConcentricCircleMeshGenerator.C:662]: giphy

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**